bash_completion

```diff
@@ -1480,13 +1528,18 @@ _available_interfaces()
 }
 
 # Echo number of CPUs, falling back to 1 on failure.
-# TODO:API: rename per conventions, rework to use vars rather than outputting
-_ncpus()
+# @var[out] ret
+_comp_get_ncpus()
 {
     local var=NPROCESSORS_ONLN
     [[ $OSTYPE == *@(linux|msys|cygwin)* ]] && var=_$var
-    local n=$(getconf $var 2>/dev/null)
-    printf %s "${n:-1}"
+    ret=$(getconf $var 2>/dev/null)
+
+    # make sure that it only contains number so that it can be safely used in
+    # comgpen -W "...".
```

Comment:
Actually, as far as `getconf` returns the expected result, it doesn't contain other than numbers.

This change actually overlaps with the changes for `-W "$(command)"` related to https://github.com/scop/bash-completion/issues/958#issuecomment-1537214890, https://github.com/scop/bash-completion/pull/961#discussion_r1186844166, https://github.com/scop/bash-completion/pull/972#discussion_r1188093136. I wouldn't like to check each case of `$(command)` whether it never outputs dangerous contents containing shell expansions, so I have been sanitizing everything that is used in `-W`.

I think I can drop the check for this particular case because we seem to be able to assume that the `getconf` output would not contain dangerous outputs regardless of the user's setup.

Reply:
I dropped the check.